server/api/orders.js

```diff
@@ -1,72 +1,150 @@
-const router = require('express').Router()
-const { models: { Order }} = require('../db')
-const { verifyInteger, 
-        verifyIsAdmin, 
-        verifyIsSpecificUserOrAdmin } = require("./apiHelpers")
-
-
-
-const PUBLIC_ORDER_FIELDS = ['total', 'email', 'addressLine1', 'addressLine2', 'city', 'zipCode']
+const router = require('express').Router();
+const {
+  models: { Order, OrderItem, Product, User },
+} = require('../db');
+const {
+  verifyInteger,
+  verifyIsAdmin,
+  verifyIsSpecificUserOrAdmin,
+} = require('./apiHelpers');
 
+const PUBLIC_ORDER_FIELDS = [
+  'total',
+  'email',
+  'addressLine1',
+  'addressLine2',
+  'city',
+  'zipCode',
+];
 
 // ADMIN only
 router.get('/', async (req, res, next) => {
   try {
     const orders = await Order.findAll({
-      attributes: PUBLIC_ORDER_FIELDS
-    })
-    res.json(orders)
+      attributes: PUBLIC_ORDER_FIELDS,
+    });
+    res.json(orders);
+  } catch (err) {
+    next(err);
+  }
+});
+
+// USER fetch the order for a given user, including its associated orderItems
+router.get('/:id', async (req, res, next) => {
+  try {
+    const { id } = req.params;
+    const orders = await Order.findAll({
+      where: {
+        userId: id,
+      },
+      include: OrderItem,
+    });
+    res.json(orders);
   } catch (err) {
-    next(err)
+    next(err);
   }
-})
+});
 
 // Specific user only
 router.post('/', async (req, res, next) => {
+  // Determine whether user has an order already and add appropriately to Order and OrderItem models
+  // First time
+  // Second time, new product
+  // Second time, previous product
   try {
-    const order = req.body
-    await Order.create(order)
-    res.status(201).send()
+    const { userId, productId, quantity } = req.body;
+    // Check to see if an Order exists or create it and associate with a user
+    const [order, createdOrder] = await Order.findOrCreate({
+      where: {
+        userId: userId,
```

Comment:
will want to filter where: {..., active: true} or something to that effect

Reply:
won't this return all orders a user has made in the past?

Reply:
ya it will... i had added "orderPaid: false" to the where statement but I must have foolishly removed it when i was refactoring... adding back.